go/libraries/doltcore/merge/constraint_violations.go

```diff
@@ -0,0 +1,507 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package merge
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"time"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/diff"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/row"
+	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
+	"github.com/dolthub/dolt/go/libraries/doltcore/table"
+	"github.com/dolthub/dolt/go/libraries/doltcore/table/editor"
+	"github.com/dolthub/dolt/go/libraries/doltcore/table/typed/noms"
+	"github.com/dolthub/dolt/go/store/types"
+)
```

Comment:
There are a lot of objects used in this calculation.  For example:

parentFkConstraintViolations(
				ctx,
				foreignKey,
				postChildTable,
				preParentSch, postParentSch, postChildSch,
				postParentIndex, postChildIndex,
				preParentData, postParentData, postChildData, postParentIndexData, postChildIndexData)

Could possibly clean this up and get rid of some code duplication by creating a class that groups some of these elements together.